packages/expo-modules-core/android/build.gradle

```diff
@@ -333,13 +338,15 @@ afterEvaluate {
 }
 
 tasks.whenTaskAdded { task ->
-  if (task.name.contains('externalNativeBuild') || task.name.startsWith('configureCMake')) {
+  if (!task.name.contains("Clean") && (task.name.contains('externalNativeBuild') || task.name.startsWith('configureCMake'))) {
     task.dependsOn(extractAARHeaders)
     task.dependsOn(extractJNIFiles)
     if (REACT_NATIVE_BUILD_FROM_SOURCE) {
-      task.dependsOn(":ReactAndroid:packageReactNdkLibs")
+      def buildType = task.name.endsWith('Debug') ? 'Debug' : 'Release'
```

Comment:
You could have lift it outside `if`s scoped and therefore prevent duplication with line 349 👀